site/en/codelabs/esp-openthread-hardware/index.lab.md

```diff
@@ -0,0 +1,600 @@
+
+> aside positive
+>
+> **Note:** For the ESP-IDF branch, We highly recommend using version 5.1.1 or later.
+
+2. Clone the ESP Thread Border Router SDK.
+
+[ESP-THREAD-BR](https://github.com/espressif/esp-thread-br) is the official ESP Thread Border Router SDK. It supports all fundamental network features to build a Thread Border Router and integrates rich product level features for quick productization.
+
+   ```console
+   $ cd <your-local-workspace>
+   $ git clone --recursive https://github.com/espressif/esp-thread-br.git
+   ```
+
+> aside positive
+>
+> **Note:** This document will show how to run this codelab on Linux. For other operating system or IDE, you can modify the relevant commands as needed.
+
+## Build and flash
+
+Duration: 5:00
+
+For build and flash ot-cli-ftd binary file on ESP boards with IEEE 802.15.4 module, you can refer to ESP-IDF example [ot_cli](https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_cli) for more details:
+
+```console
+$ cd <your-idf-path>/examples/openthread/ot_cli
+$ idf.py set-target <your-board-type>
+```
+
+Enable the joiner feature via menuconfig:
+
+```console
+$ idf.py menuconfig
+```
+
+Component config > OpenThread > Enable Joiner, then build and flash.
+
+> aside positive
+>
+> **Note:** Connect `UART` port on the ESP boards to the host. To show the serial port number of the ESP boards, you can use `$ ls /dev/ttyUSB*`.
+
+```console
+$ idf.py -p <your-local-port> build flash monitor
+```
+
+For building and flashing the ot-br binary file on ESP Thread Border Router Board, you need to build the RCP binary file first. This RCP binary file doesn't need to be explicitly flashed to the device on the ESP Thread Border Router Board. It will be included in the Border Router binary file and flashed to the ESP32-H2 chip upon first boot(or the RCP firmware changed). You can refer to [ESP thread BR docs](https://docs.espressif.com/projects/esp-thread-br/en/latest/dev-guide/build_and_run.html) for more details:
+
+```console
+$ cd <your-idf-path>/examples/openthread/ot_rcp
+$ idf.py set-target esp32h2
+$ idf.py build
+$ cd <your-esp-thread-br-path>/examples/basic_thread_border_router
+$ idf.py set-target esp32s3
+```
+
+Enable the commissioner feature via menuconfig:
+
+```console
+$ idf.py menuconfig
+```
+
+Component config > OpenThread > Enable Commissioner, then build and flash.
+
+> aside positive
+>
+> **Note:** Only the `USB2` port on the ESP Thread Border Router Board needs to be connected to the host. To show the serial port number of the ESP Thread Border Router Board, you can use `$ ls /dev/ttyACM*`.
+
+```console
+$ idf.py -p <your-local-port> build flash monitor
+```
+
+## Forming a Thread network on the Thread Border Router
+
+Duration: 3:00
+
+Now you can form a Thread network using the OpenThread command line on ESP Thread Border Router Board(BR Commissioner):
```

Comment:
plz add space before (

Reply:
Additional note: Please update the following file to add this codelab:

https://github.com/openthread/ot-docs/blob/main/site/en/guides/_toc.yaml

thanks!